azkaban-webserver/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java

```diff
@@ -563,29 +566,38 @@ private void handlePurgeProject(HttpServletRequest req,
                                                            // from ints)
       }
 
-      // invalid project
-      if (project == null) {
-        ret.put("error", "invalid project");
-        isOperationSuccessful = false;
-      }
+      if (!projectActive) {
+        // invalid project
+        if (project == null) {
+          ret.put("error", "invalid project");
+          isOperationSuccessful = false;
+        }
 
-      // project is already deleted
-      if (isOperationSuccessful
-          && projectManager.isActiveProject(project.getId())) {
-        ret.put("error", "Project " + project.getName()
-            + " should be deleted before purging");
-        isOperationSuccessful = false;
-      }
+        // project is already deleted
+        if (isOperationSuccessful && projectManager.isActiveProject(project.getId())) {
+          ret.put("error", "Project " + project.getName() + " should be deleted before purging");
+          isOperationSuccessful = false;
+        }
 
-      // only eligible users can purge a project
-      if (isOperationSuccessful && !hasPermission(project, user, Type.ADMIN)) {
-        ret.put("error", "Cannot purge. User '" + user.getUserId()
-            + "' is not an ADMIN.");
-        isOperationSuccessful = false;
-      }
+        // only eligible users can purge a project
+        if (isOperationSuccessful && !hasPermission(project, user, Type.ADMIN)) {
```

Comment:
This logic seems to be common in both cases. Right? If so, do it in a common place?